app/src/main/java/com/example/android/architecture/blueprints/todoapp/tasks/TasksViewModel.kt

```diff
@@ -72,6 +71,10 @@ class TasksViewModel @Inject constructor(
         combine(tasksRepository.getTasksStream(), _savedFilterType) { tasks, type ->
             filterTasks(tasks, type)
         }
+            .catch {
+                showSnackbarMessage(R.string.loading_tasks_error)
```

Comment:
Ditto. This looks like a smell. `catch` shouldn't create a side effect that updates the ui state. It should notify the collector appropriately. Instead of an `Async.Success`, it should probably be an `Async.Error` if we really care about that error message.

It doesn't seem like that snackbar was shown before when the Result was an error, right?

Reply:
> It doesn't seem like that snackbar was shown before when the Result was an error, right?

The snackbar was shown previously when it was `Result.Error`: https://github.com/android/architecture-samples/pull/906/files#diff-631edb6efa01767b52856a0e498dc3987e8a2f0e8ba02837377dfbc5684af0cfL153

I'll create an `Async.Error` to handle this.